pkg/azuredisk/azure_controller_common.go

```diff
@@ -190,38 +190,54 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, diskName, diskURI str
 		return -1, err
 	}
 
-	var waitForDetachHappened bool
-	if c.WaitForDetach {
-		// wait for disk detach to finish first on the same node
-		if err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 30*time.Second, true, func(context.Context) (bool, error) {
-			detachDiskReqeustNum, err := c.getDetachDiskRequestNum(node)
-			if err != nil {
-				return false, err
-			}
-			if detachDiskReqeustNum == 0 {
-				return true, nil
-			}
-			klog.V(4).Infof("there are still %d detach disk requests on node %s, wait for detach to finish, current disk: %s", detachDiskReqeustNum, node, diskName)
-			waitForDetachHappened = true
-			return false, nil
-		}); err != nil {
-			klog.Errorf("current disk: %s, wait for detach disk requests on node %s failed: %v", diskName, node, err)
-		}
-	}
-
 	c.lockMap.LockEntry(node)
 	defer c.lockMap.UnlockEntry(node)
 
-	if !waitForDetachHappened && c.AttachDetachInitialDelayInMs > 0 && requestNum == 1 {
+	if c.AttachDetachInitialDelayInMs > 0 && requestNum == 1 {
 		klog.V(2).Infof("wait %dms for more requests on node %s, current disk attach: %s", c.AttachDetachInitialDelayInMs, node, diskURI)
 		time.Sleep(time.Duration(c.AttachDetachInitialDelayInMs) * time.Millisecond)
 	}
 
+	numDisksAllowed := math.MaxInt
+	if c.CheckDiskCountForBatching {
+		_, instanceType, err := getNodeInfoFromLabels(ctx, string(nodeName), c.cloud.KubeClient)
+		if err != nil {
+			return -1, err
```

Comment:
if there is error, just error log here, and exit CheckDiskCountForBatching since `getNodeInfoFromLabels` may fail in many ways, e.g. lable does not exist or kubeclient may not have permission, etc.